service/src/main/java/bio/terra/workspace/app/controller/ReferencedGcpResourceController.java

```diff
@@ -753,4 +756,16 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
             .effectiveCloningInstructions(effectiveCloningInstructions.toApiModel());
     return new ResponseEntity<>(result, HttpStatus.OK);
   }
+
+  private AuthenticatedUserRequest getPetRequest(UUID workspaceId) {
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    return petSaService
+        .getWorkspacePetCredentials(workspaceId, userRequest)
+        .orElseThrow(
+            () ->
+                new BadRequestException(
+                    String.format(
+                        "Pet SA credentials not found for user %s on workspace %s",
+                        userRequest.getEmail(), workspaceId)));
+  }
```

Comment:
per slack discussion, this won't work for references. The underlying `getWorkspacePetCredentials` tries to create a pet in the GCP context of the provided workspace, but we may not have a GCP cloud context when we're working with references. Instead of throwing here, I think it's reasonable to fall back to end-user credentials (i.e.
```
.getWorkspacePetCredentials(workspaceId, userRequest).orElse(userRequest)
```
and renaming this method to something like `maybeGetPetRequest` or `getCloningCredentials` with a comment explaining why how it works.